src/Scheduler/GenericScheduler.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console\Scheduler;
+
+use DatePeriod;
+use DateTime;
+use DateInterval;
+use DateTimeImmutable;
+use Tempest\Console\ConsoleCommand;
+use Tempest\Console\ConsoleConfig;
+use Tempest\Console\ConsoleOutput;
+
+final class GenericScheduler implements Scheduler
+{
+    private string $path;
+    private DateTime $dayStart;
+    private DateTime $now;
+
+    public function __construct(
+        private ConsoleConfig $config,
+        private ConsoleOutput $output,
+    ) {
+        $this->path = "php tempest"; // todo: discover path
+    }
+
+    public function run(?DateTime $date = null): void
+    {
+        $now = $date ?? (new DateTime());
+
+        $this->dayStart = (clone $now)->setTime(0, 0, 0);
+        $this->now = ($now)->setTime(
+            (int) $now->format('H'),
+            (int) $now->format('i'),
+        );
+
+        $commands = $this->getCommandsToRun();
```

Comment:
I don't like this approach of setting instance variables in one function that influence the behaviour of other functions. I think the better approach is to pass these variables explicitly to `getCommandsToRun`